rust/chains/hyperlane-cosmos/src/merkle_tree_hook.rs

```diff
@@ -309,15 +309,18 @@ impl Indexer<MerkleTreeInsertion> for CosmosMerkleTreeHookIndexer {
             .await
             .into_iter()
             .flatten()
-            .filter_map(|(logs_res, block_number)| match logs_res {
-                Ok(logs) => Some(logs),
-                Err(err) => {
+            .map(|(logs, block_number)| {
+                if let Err(err) = &logs {
                     warn!(?err, ?block_number, "Failed to fetch logs for block");
-                    None
                 }
+                logs
             })
+            // Propagate errors from any of the queries. This will cause the entire range to be retried,
```

Comment:
since we propagate errors from indexing, the contract_sync actually gets to sleep [here](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/816924c388a15a04a68f67fd292383789c04cb94/rust/hyperlane-base/src/contract_sync/mod.rs#L160).

You can see that before I deployed this ([logs here](https://cloudlogging.app.goo.gl/k13MfkccZFPPgHud6)), that warning wasn't logged at all, so we'd just query the same range in a loop